### PR TITLE
add rubyrep

### PR DESCRIPTION
**Pull request by izakp**
Setting up [rubyrep](http://www.rubyrep.org/index.html) with JRuby (uses the JDBC Postgres drivers installed and claims to be faster)

Will set up a deployment in Convection staging to test a sync and then streaming replication to a new database.

cc @ashkan18 hoping this is a nice way to migration Radiation as well!

### CODE REVIEW

**Comment by ashkan18** [comment]
🤓 thats ridiculous !!
image

**Comment by izakp** [comment]
Well, let's see if it all works ;)  Since it creates triggers in the "left" or source database during [replication](http://www.rubyrep.org/replication.html) as well as adjusts sequences to avoid auto-incrementing ID conflicts (unless disabled) I'm going to download a backup and try to set up replication between two local databases to make sure there are no destructive actions on the source DB

**Comment by izakp** [comment]
I'm finding a lot of problems with this tool.  It doesn't appear to be well maintained... handle foreign key constraints in replicated tables well, or NULLs in rows either...

**Comment by izakp** [comment]
Aha - so the foreign key constrains can be avoided by using [before_sync and after_sync hooks](https://www.rubyrep.org/sync_hooks.html) which require superuser permissions on the downstream ("right") database.  See [this gist](https://gist.github.com/alehmann/316522/05a8be062640d6eb2d9a7db05e58e8508f1b4caf#file-postgres-syncing-tables-with-circular-foreign-key-constraints-L38) for an example.

The NULL errors I encountered were due to the fact that I didn't completely migrate the downstream ("right") database schema to be up-to-date with the upstream ("left") database

**Comment by izakp** [comment]
So before replication, we must run this commend on RDS for the user `db_user`:

`ALTER USER db_user WITH SUPERUSER;`

Then after:

`ALTER USER db_user WITH NOSUPERUSER;`

**Comment by izakp** [comment]
Ok so I was able to test a local postgres sync of Convection's database with this `default.conf` (after importing the staging database to the local `convection` database, then setting up the `convection_sync` database with superuser permissions as described above and running `rake db:migrate`

```
RR::Initializer::run do |config|
  config.left = {
    :adapter  => 'postgresql',
    :database => 'convection',
    :username => 'convection',
    :password => 'convection',
    :host     => '127.0.0.1'
  }

  config.right = {
    :adapter  => 'postgresql',
    :database => 'convection_sync',
    :username => 'convection',
    :password => 'convection',
    :host     => '127.0.0.1'
  }

  config.include_tables /./

  config.options[:adjust_sequences] = false

  config.add_table_options 'ar_internal_metadata',
    :before_table_sync => "alter table ar_internal_metadata disable trigger all",
    :after_table_sync => "alter table ar_internal_metadata enable trigger all"

  config.add_table_options 'assets',
    :before_table_sync => "alter table assets disable trigger all",
    :after_table_sync => "alter table assets enable trigger all"

  config.add_table_options 'offers',
    :before_table_sync => "alter table offers disable trigger all",
    :after_table_sync => "alter table offers enable trigger all"

  config.add_table_options 'partner_submissions',
    :before_table_sync => "alter table partner_submissions disable trigger all",
    :after_table_sync => "alter table partner_submissions enable trigger all"

  config.add_table_options 'partners',
    :before_table_sync => "alter table partners disable trigger all",
    :after_table_sync => "alter table partners enable trigger all"

  config.add_table_options 'submissions',
    :before_table_sync => "alter table submissions disable trigger all",
    :after_table_sync => "alter table submissions enable trigger all"

  config.add_table_options 'users',
    :before_table_sync => "alter table users disable trigger all",
    :after_table_sync => "alter table users enable trigger all"

end
```

Then the `sync` command copied over all but the one row in `ar_internal_metadata` and replication appeared to be active

File: rubyrep/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM jruby:9.0
+
+RUN curl -L https://bintray.com/rubyrep/rubyrep/download_file?file_path=rubyrep-2.0.1.zip -o /tmp/rubyrep-2.0.1.zip
+
+RUN unzip /tmp/rubyrep-2.0.1.zip -d /
```
Comment (izakp) [inline rubyrep/Dockerfile]: I guess I can delete the archive here

**Comment by izakp** [comment]
Idk if we can access some other list of tables to DRY up these explicit table triggers, but at least it's working

**Pull request merged**